app/controllers/api/rss_channels_controller.rb

```diff
@@ -11,6 +11,11 @@ def create
       head :created
     end
 
+    def destroy
+      @url = Post.find_by(url: params[:url])
+      @url.destroy
```

Comment:
```suggestion
      @channel.destroy!
```

これは rails に慣れてないとわからないのですが、 `!` がないと削除に失敗した場合に例外が発生しません。
`!` をつけると例外が発生し、必要なHTTP Status を返してくれます（多分）。
ここでも